src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -365,17 +365,22 @@ public DetailedPackageMetadata PackageMetadata
                 {
                     _packageMetadata = value;
 
+                    string newDeprecationReasons = null;
+                    string newAlternatePackageText = null;
                     if (_packageMetadata?.DeprecationMetadata != null)
                     {
-                        PackageDeprecationReasons = ExplainPackageDeprecationReasons(_packageMetadata.DeprecationMetadata.Reasons?.ToList());
```

Comment:
You can also use an else clause, to avoid the extra allocation.

Reply:
Initially I was going to do this but, I quickly realized that there would have to be two `else` clauses:
```
if (_packageMetadata?.DeprecationMetadata != null)
{
    PackageDeprecationReasons = ExplainPackageDeprecationReasons(_packageMetadata.DeprecationMetadata.Reasons?.ToList());

    var alternatePackage = _packageMetadata.DeprecationMetadata.AlternatePackage;
    if (alternatePackage != null)
    {
        PackageDeprecationAlternatePackageText = GetPackageDeprecationAlternatePackageText(alternatePackage);
    } 
    else 
    {
        PackageDeprecationAlternatePackageText = null;
    }
}
else
{
    PackageDeprecationReasons = null;
    PackageDeprecationAlternatePackageText = null;
}
```

I felt the allocation was simpler and easier to maintain.